_posts/2024-01-08-windows-support-1.11.md

```diff
@@ -0,0 +1,53 @@
+
+## Playground support
+
+The playground now works on Windows. For a long time, `Process#wait` was synchronous on Windows, which means other fibers would not get the chance to run until the process exits. The playground server communicates with the compiled program via multiple WebSocket messages, and a synchronous `Process#wait` would effectively block the server. This is no longer the case since [#13908], and you will be able to use `crystal play` just like on other systems.
+
+## Finishing DLL coverage
+
+Currently, the compiler is statically linked for two reasons: the C function `LibC.snprintf` is not available in the C runtime DLLs, and the compiler depends on LLVM's C++ API, which requires a static LLVM build tree.
+
+The main remaining use of `LibC.snprintf` was implementing the floating-point format specifiers for Crystal's `sprintf` and `String#%`, such as `%.2f` and `%g`. By porting the Ryu Printf algorithm to native Crystal ([#14067], [#14084], [#14102], [#14123], [#14132]), this dependency is now gone from Windows and Unix-like systems, with the added benefit that float printing is unaffected by the active C locale.
+
+There is nothing Crystal could do if the used LLVM version does not expose all the needed functionality via LLVM's C API, so instead the Crystal team now upstreams those necessary APIs to LLVM itself. After [#14082] and [#14101], the `LLVM-C.dll` library from LLVM 18 or above plus a small configuration file is sufficient; the compiler will eventually become dynamically linked, and still be able to rebuild itself without an LLVM build tree.
+
+## `dll` parameter for `@[Link]`
+
+Crystal originally used a technique called DLL delay-loading to load DLLs from non-default locations, but it was revealed that not all DLLs work that way, so Crystal is now trying a different strategy to simplify dynamic builds. The `@[Link]` annotation now accepts a `dll` parameter which lets you specify non-system DLL dependencies: ([#14131])
+
+```crystal
+@[Link("z", dll: "zlib1.dll")]
+lib LibYAML
+end
+```
+
+The compiler will look up the DLLs in a set of search paths, and copy them to the same directory as the built program, including the temporary executables built for commands like `crystal run`. Refer to the [documentation](https://crystal-lang.org/api/1.11.0/Link.html) for more details.
+
+## Interpreter support
+
+With everything combined, the interpreter will now more or less run on Windows. There are still some rough edges pending resolution before the interpreter is released on Windows, but you could try rebuilding the compiler itself to see the REPL session in action.
```

Comment:
Is there a reason to not update our CI to build the .exe with interpreter support, even if very rough? Why ask people to do it?

Reply:
At least one recent PR didn't make it into 1.11 (https://github.com/crystal-lang/crystal/pull/14146), there is no need to rush

Reply:
I think we could start enabling it.